internal/webhook/external/github.go

```diff
@@ -0,0 +1,147 @@
+package external
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+
+	gh "github.com/google/go-github/v71/github"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	xhttp "github.com/akuity/kargo/internal/http"
+	"github.com/akuity/kargo/internal/logging"
+)
+
+// githubHandler handles push events for github.
+// After the request has been authenticated,
+// the kubeclient is queried for all warehouses that contain a subscription
+// to the repo in question. Those warehouses are then patched with a special
+// annotation that signals down stream logic to refresh the warehouse.
+func githubHandler(c client.Client) http.HandlerFunc {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		logger := logging.LoggerFromContext(ctx)
+		logger.Debug("identifying source repository")
+
+		secret, ok := os.LookupEnv("GH_WEBHOOK_SECRET")
+		if !ok {
+			logger.Error(
+				errors.New("GH_WEBHOOK_SECRET unset"),
+				"environment misconfiguration",
+			)
+			xhttp.WriteErrorJSON(w,
+				xhttp.Error(
+					// keep it vague, no need to leak config details
+					errors.New("unexpected server error"),
+					http.StatusInternalServerError,
+				),
+			)
+			return
+		}
```

Comment:
Checking this on every request seems unnecessary. Normally, I'd rather handle this up front as things are initializing, but we know this whole getting the secret from an env var business is temporary because we'll eventually be getting the secret from a `Secret` referenced by a `WebhookReceiver`... so the best thing to do to clean this up here is probably to remove this and temporarily hard code a secret with a TODO comment about getting the secret from the CRD.